spec/unit/worker_async_spec.rb

```diff
@@ -43,6 +44,54 @@ module Backend
             expect(test_job).to have_received(:fail).with(error)
           end
         end
+
+        describe 'when there are pending jobs in the memory queue' do
+          subject do
+            WorkerAsync.new
+          end
+
+          let(:test_job) { instance_double('BackgroundJob') }
+          let(:n_reports) { 10 }
+          let(:queue) { Queue.new.tap { |q| n_reports.times.map { q.push test_job } } }
+
+          before do
+            subject.instance_variable_set(:@jobs, queue)
+          end
+
+          it 'clears the queue'do
+            expect(subject).to receive(:perform).exactly(n_reports).times.with(test_job)
+            queue.close
+
+            subject.send(:clear_queue)
+          end
+
+          it 'clears the queue also when new jobs are added during the execution' do
+            n_new_reports = 5
+            expect(subject).to receive(:perform).exactly(n_reports + n_new_reports).times.with(test_job)
+
+            # The thread will process all jobs in the queue and wait forever until the queue is closed
+            thread = Thread.new { Sync { subject.send(:clear_queue) } }
+            Sync do
+              barrier = Async::Barrier.new
+              n_new_reports.times { barrier.async { queue.push test_job } }
+              barrier.wait # We don't want to close the queue while there are still tasks pushing jobs to it
+            end
+            queue.close # Unlock the thread
+
+            # Interrupt the test if the thread is locked for more than 10 seconds
+            # We assume something went wrong
+            t_start = Time.now
+            while thread.alive?
+              if Time.now - t_start > 10
```

Comment:
another nitpick, to use monotonic clock for such purposes

```suggestion
            t_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
            while thread.alive?
              if Process.clock_gettime(Process::CLOCK_MONOTONIC) - t_start > 10
```